src/js/API/wall/methods.ts

```diff
@@ -37,14 +37,30 @@ export const createNewPost = async (content: string): Promise<void> => {
   );
 };
 
-export const createNewPostWithPhoto = async (
+export const createNewPostWithPhotoAndPeak = async (
   content: string,
-  file: File
+  file: File | null,
+  latitude: number | null,
+  longitude: number | null,
+  peakId: string | null
 ): Promise<void> => {
   const body = new FormData();
   body.append('content', content);
-  body.append('file', file);
-  await axios.post(createNewPhotoPostUrl(), body, {
-    headers: authHeader(),
-  });
+  if (file) body.append('file', file);
+  if (latitude) body.append('latitude', latitude.toString());
+  if (longitude) body.append('longitude', longitude.toString());
+
+  await axios.post(
+    peakId
```

Comment:
Ta logika wysyłania requestów jest obrzydliwa. Szkoda że musimy robić to w taki sposób 😢

Reply:
Niestety, w sumie nie dużo by wymagało żeby to zmienić na backu, ale niech już zostanie.